src/AutoMapper/MapperConfiguration.cs

```diff
@@ -37,7 +37,8 @@ public MapperConfiguration(MapperConfigurationExpression configurationExpression
             Configuration = new ProfileMap(configurationExpression);
             Profiles = new[] { Configuration }.Concat(configurationExpression.Profiles.Select(p => new ProfileMap(p, configurationExpression))).ToArray();
 
-            configurationExpression.Advanced.BeforeSeal?.Invoke(this);
+            foreach (var beforeSealAction in configurationExpression.Advanced.BeforeSealActions)
+                beforeSealAction?.Invoke(this);
```

Comment:
I think this code should be inside Advanced and then you don't need to expose the list.